File: src/web/llm/__smoke__/gateway.smoke.ts
/**
 * gateway.smoke.ts — end-to-end smoke test for the LiteLLM gateway.
 *
 * Run:
 *   npm run llm:smoke
 *
 * Expected output:
 *   ✓ chat completion returned N chars
 *   ✓ embedding returned <dim> floats, first value = ...
 *
 * Requires:
 *   - `npm run llm:up` (docker-compose stack running)
 *   - .env with LITELLM_BASE_URL + LITELLM_API_KEY
 */

import { chatText, isGatewayConfigured } from "../gateway";
import { generateEmbedding } from "../embeddings";
import { config } from "../../../config";

async function main() {
  console.log("LLM Gateway smoke test");
  console.log("-----------------------");
  console.log(`baseUrl        : ${config.llm.baseUrl}`);
  console.log(`chatModel      : ${config.llm.chatModel}`);
  console.log(`embeddingModel : ${config.llm.embeddingModel}`);
  console.log(`embeddingDim   : ${config.llm.embeddingDim}`);
  console.log(`apiKey set?    : ${config.llm.apiKey ? "yes" : "NO"}`);
  console.log("");

  if (!isGatewayConfigured()) {
    console.error("✗ Gateway is not configured (missing LITELLM_API_KEY?)");
    process.exit(1);
  }

  // 1. Chat
  console.log("→ Testing chat completion...");
  const t0 = Date.now();
  const chatOut = await chatText({
    system: "You are a helpful assistant. Reply in one short sentence.",
    user: "Say hello in Traditional Chinese.",
    maxTokens: 60,
  });
  const chatMs = Date.now() - t0;
  if (!chatOut) {
    console.error("✗ chat completion returned null");
    process.exit(1);
  }
  console.log(`✓ chat OK (${chatMs}ms): ${chatOut}`);
  console.log("");

  // 2. Embedding
  console.log("→ Testing embedding...");
  const e0 = Date.now();
  const vec = await generateEmbedding("鑽石等級賭客偏好百家樂", "document");
  const embMs = Date.now() - e0;
  const nonZero = vec.some((v) => v !== 0);
  if (!nonZero) {
    console.error("✗ embedding returned a zero vector");
    process.exit(1);
  }
  console.log(
    `✓ embedding OK (${embMs}ms): dim=${vec.length}, first=${vec[0].toFixed(6)}, last=${vec[vec.length - 1].toFixed(6)}`
  );
  console.log("");
  console.log("All checks passed.");
}

main().catch((err) => {
  console.error("Smoke test failed:", err);
  process.exit(1);
});

File: server/app/smoke/gateway_smoke.py
"""Smoke test: verify LiteLLM chat + voyage local embeddings both work.

Run with::

    uv run python -m app.smoke.gateway_smoke
"""

from __future__ import annotations

import asyncio
import sys

from app.llm.embeddings import generate_embedding
from app.llm.gateway import chat_text, is_gateway_configured


async def _smoke() -> int:
    ok = True

    print("[smoke] gateway configured:", is_gateway_configured())
    if is_gateway_configured():
        text = await chat_text(
            system="You are a terse assistant.",
            user="Reply with exactly the token PONG.",
            temperature=0.0,
            max_tokens=10,
        )
        print("[smoke] chat_text →", text)
        if not text:
            ok = False
    else:
        print("[smoke] skipping chat_text (LITELLM_API_KEY not set)")

    vec = await generate_embedding("Hello, world!", "document")
    non_zero = sum(1 for v in vec if v != 0)
    print(f"[smoke] embedding dim={len(vec)} non-zero={non_zero}")
    if non_zero == 0:
        print("[smoke] WARNING: embedding is all zeros — voyage local unavailable")
        ok = False

    return 0 if ok else 1


if __name__ == "__main__":
    sys.exit(asyncio.run(_smoke()))
